mlrun/run.py

```diff
@@ -738,6 +738,9 @@ def update_common(fn, spec):
         filename, handler = ApplicationRuntime.get_filename_and_handler()
 
     is_nuclio, sub_kind = RuntimeKinds.resolve_nuclio_sub_kind(kind)
+
+    mlrun.utils.helpers.validate_handler_name(function_kind=sub_kind, handler=handler)
+
```

Comment:
1. This should be below around:
```
    if is_nuclio:
        runtime = RuntimeKinds.resolve_nuclio_runtime(kind, sub_kind)
        # default_handler is only used in :mlrun sub kind, determine the handler to invoke in function.run()
        runtime.spec.default_handler = handler if sub_kind == "mlrun" else ""
```
2. `code_to_function` is not the only way to create a function. What about `new_function`?

in `new_function` you need to validate the `handler` param as you did here but `new_function` is also used when importing a function. i.e. a function can be created from a dict or a yaml and in that case it will bypass this validation. To handle that you need to validate in the server - in `pre_deploy_validation`

Reply:
When creating a function with `new_function`, we cannot set the function kind to `nuclio:mlrun`, It raises an error:

> Unsupported runtime (nuclio:mlrun) or missing command, supported runtimes: remote,nuclio,serving,dask,job,spark,remote-spark,mpijob,local,databricks,application

Reply:
Or we want to enable it and add the nuclio:mlrun kind to RuntimeKinds

Reply:
I added the validation above if `is_nuclio:` so that we can use this function in the future to check the handler name for other function kinds, not just for Nuclio. Additionally, this allows us to fail earlier, before building the file.

Reply:
walla. no need to support it then if its not already supported. thanks for explaining 😄